### PR TITLE
Add snow specific Bottlerocket settings: admin, control, bootstrap snow containers

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

#4050 

Base on #4208 

*Description of changes:*

- Override default BR admin, control container images URL
- Add bottlerocket-bootstrap-snow to custom bootstrap container image

Hold till #4208 is merged first.

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -209,5 +209,5 @@ replace (
 	github.com/opencontainers/runc => github.com/opencontainers/runc v1.1.2
 
 	// need the modifications eksa made to the capi api structs
-	sigs.k8s.io/cluster-api => github.com/abhay-krishna/cluster-api v1.2.0-custom.rc2
+	sigs.k8s.io/cluster-api => github.com/jiayiwang7/cluster-api v0.0.0-20221123142628-4b8c7b3f61eb
```
Comment (jiayiwang7) [inline go.mod:212]: discard this change when https://github.com/abhay-krishna/cluster-api/pull/3 is merged

File: config/crd/bases/anywhere.eks.amazonaws.com_bundles.yaml [hunk 2]
```diff
@@ -216,7 +216,7 @@ spec:
                       - metadata
                       - version
                       type: object
-                    bottlerocketAdmin:
+                    bottlerocket:
```
Comment (vivek-koppuru) [inline config/crd/bases/anywhere.eks.amazonaws.com_bundles.yaml:219]: If we upgrade an older cluster to a newer one with these changes and the older bundle is referenced, would this be a problem? Do we need to phase it out instead?

File: pkg/providers/snow/apibuilder.go [hunk 3]
```diff
@@ -235,3 +242,24 @@ func SnowMachineTemplate(name string, machineConfig *v1alpha1.SnowMachineConfig)
 		},
 	}
 }
+
+func bottlerocketBootstrapSnow(image releasev1.Image) bootstrapv1.BottlerocketBootstrapContainer {
```
Comment (vivek-koppuru) [inline pkg/providers/snow/apibuilder.go:246]: Why is this here and not in the clusterapi package? Do we want a bootstrap file for this instead?

Comment (jiayiwang7) [reply pkg/providers/snow/apibuilder.go:246]: bc it's a snow specific bootstrap container including the settings like name and mode. I can move it to another file

File: config/crd/bases/anywhere.eks.amazonaws.com_bundles.yaml (hunk 2 above)
Comment (jiayiwang7) [reply config/crd/bases/anywhere.eks.amazonaws.com_bundles.yaml]: I don't think it would be an issue: we have the core components - eksa components upgrade in the upgrade workflow. This will update the eksa CRD (including the bundle CRD) and apply the new bundle spec, before the actual cluster upgrade.

**Review by vivek-koppuru — APPROVED** [review]
